manual/opt.tex

```diff
@@ -153,9 +153,110 @@ \subsubsection{quartic}
+For example, suppose the initial 
+trial wave function's parameters are taken as a guess for the first excited state.
+Then, by setting $\omega$ to be a value in between the ground state and first excited state 
+energies would cause the trial wave function to be optimized for the first excited state. 
+
+When the trial wave function contains more than tens of thousands of parameters, constructing and storing the linear method matrices may become a memory bottleneck. 
+To avoid explicit
+construction of these matrices, the adaptive optimizer implements the recently developed ``block linear method''.  The block linear method (BLM) tries to find an approximate 
+solution $\vec{c}_{opt}$ to the standard LM generalized eigenvalue problem by (a) dividing the variable space into a number of blocks (b) making intelligent estimates for which 
+directions within those blocks will be most important for constructing $\vec{c}_{opt}$, and (c) estimating $\vec{c}_{opt}$ by solving a smaller, more memory-efficient 
+eigenproblem in the basis of these supposedly important block-wise directions. 
+
+\begin{table}[h]
+\begin{center}
+\begin{tabularx}{\textwidth}{l l l l l l }
+\hline
+\multicolumn{6}{l}{\texttt{linear} method} \\
+\hline
+\multicolumn{2}{l}{parameters}  & \multicolumn{4}{l}{}\\
+   &   \bfseries name     & \bfseries datatype & \bfseries values & \bfseries default   & \bfseries description \\
+   %&   \texttt{stepsize} &  real     & 0--1 & 0.25  & Step size for moving parameters\\
+   &   \texttt{max\_relative\_change} &  real     & $>0$ & 10.0 & Allowed change in cost function\\
+   &   \texttt{max\_param\_change} &  real     & $>0$ & 0.3 & Allowed change in wave function parameter\\
+   &   \texttt{shift\_i} &  real     & $>0$ & 0.01 & Initial diagonal      stabilizer added to the Hamiltonian matrix\\
+   &   \texttt{shift\_s} &  real     & $>0$ & 1.00 & Initial overlap-based stabilizer added to the Hamiltonian matrix\\
+   &   \texttt{targetExcited} &  text   & yes, no & no & Target excited state\\
+   &   \texttt{omega} &  real   & $<0$ & & Energy shift used to target different excited state\\
+   &   \texttt{chase\_lowest} &  text   & yes, no & yes & Chase the lowest eigenvector in iterative solver\\
+   &   \texttt{chase\_closest} &  text   & yes, no & no & Chase the eigenvector closest to initial guess\\
+   &   \texttt{block\_lm} &  text   & yes, no & no & Use block linear method\\
+   &   \texttt{nblocks} &  integer   & $>0$ &  & \# of blocks in BLM\\
+   &   \texttt{nolds} &  integer   & $>0$ &  & \# of old update vectors used in BLM\\
+   &   \texttt{nkept} &  integer   & $>0$ &  & \# of eigenvectors to keep per block in BLM\\
+  \hline
+\end{tabularx}
+\end{center}
+\end{table}
+
+Additional information:
+\begin{itemize}
+  \item \texttt{shift\_i}.  This is the initial value of direct term added to the diagonal of the Hamiltonian matrix. 
```

Comment:
"of the direct term"

The last sentence (both after shift_i and shift_s) - do you mean the value of shift_i/s is adjusted by the optimizer as it proceeds?  Is this implicit in describing these as 'the initial value', or is it important to understanding how the algorithm works (ie, that the value the user gives will not be the actual value used later on)